src/clarity-angular/tabs/vtabs.spec.ts

```diff
@@ -0,0 +1,182 @@
+                    if (item.activated) {
+                        activatedCount++;
+                    }
+                });
+            });
+
+            it("should emit an event when the active nav link changes", function () {
+                let links = context.testElement.querySelectorAll("li");
+                let activator: HTMLElement = links[1];
+                spyOn(context.testComponent, "handleVTabsNavChange");
+                activator.click();
+                expect(context.testComponent.handleVTabsNavChange).toHaveBeenCalled();
+            });
+
+            it("should emit an event when the active content changes", function () {
+
+                let links = context.testElement.querySelectorAll("li");
+                let activator: HTMLElement = links[1];
+                spyOn(context.testComponent, "handleVTabsContentChange");
+                activator.click();
+                expect(context.testComponent.handleVTabsContentChange).toHaveBeenCalled();
+            });
+
+            it("maps links to content by id", function () {
+                let linkArray = context.clarityDirective.vtabLinks.toArray();
+                let contentArray = context.clarityDirective.vtabContents.toArray();
+                for (let i = 0; i < linkArray.length; i++) {
+                    expect(linkArray[i].id).toEqual(contentArray[i].id);
+                }
+            });
+
+            it("activates content when link is clicked", function () {
+                let lastContent = context.clarityDirective.vtabContents.last;
+                let links = context.testElement.querySelectorAll("li");
+                let activator: HTMLElement = links[1];
+                expect(lastContent.activated).toBe(false);
+                activator.click();
+                expect(lastContent.activated).toBe(true);
+            });
+        });
+
+        describe("allows user to configure", function () {
+
+            let context: TestContext<VTabs, PreselectedVTabs>;
+            beforeEach(function () {
+                context = this.create(VTabs, PreselectedVTabs);
+            });
+
+            it("a pre-selected tab", function () {
+                let activeLink = context.testElement.querySelector(".nav-link.active");
+                expect(activeLink.textContent.trim()).toBe("The Wastelands");
+            });
+
+            it("a reversed state", function () {
+                let isReversed = context.testElement.querySelector(".vtabs.is-reversed");
+                expect(isReversed).not.toBeNull();
+            });
+        });
+    });
+}
+
+@Component({
+    template: `
+        <clr-vtabs (clrVTabsNavChange)="handleVTabsNavChange($event)"
+                   (clrVTabsNavChange)="handleVTabsContentChange($event)">
+            <clr-vtabs-nav-link>The Gunslinger</clr-vtabs-nav-link>
+            <clr-vtabs-nav-link>The Wastelands</clr-vtabs-nav-link>
+            <clr-vtabs-content>
```

Comment:
Do we plan to write a test that validates what happens when a user generates content/nav-links with ng-fors?

What about if items are hidden with ng-if?

These are all things I'm having to test in the wizard, so I'm perhaps feeling a little picked on here.